packages/forms/src/styled/input-group/StyledAppend.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import styled, { ThemeProps, DefaultTheme, css } from 'styled-components';
+import { retrieveComponentStyles, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+
+const COMPONENT_ID = 'forms.input_group.append';
+
+const borderStyles = (props: ThemeProps<DefaultTheme>) => {
+  const startDirection = props.theme.rtl ? 'right' : 'left';
+  const endDirection = props.theme.rtl ? 'left' : 'right';
+
+  return css`
+    /* stylelint-disable */
+    margin-${startDirection}: -${props.theme.borderWidths.sm};
+
+    & > * {
+      border-top-${startDirection}-radius: 0;
+      border-bottom-${startDirection}-radius: 0;
+    }
+
+    &:not(:last-child) > * {
+      border-top-${endDirection}-radius: 0;
+      border-bottom-${endDirection}-radius: 0;
+    }
+    /* stylelint-enable */
+  `;
+};
+
+export const StyledAppend = styled.div.attrs({
+  'data-garden-id': COMPONENT_ID,
+  'data-garden-version': PACKAGE_VERSION
+})`
+  display: flex;
+  z-index: 1;
+
+  &:hover,
+  &:active {
+    z-index: 2;
```

Comment:
> This was tricky to bring over, but I think I got it working. We need to start out the z-index at 1 with the InputGroup to beat the StyledTextInput styling.

I'm not seeing that `StyledTextInput` has a `z-index`.

Reply:
Without starting at `z-index: 1` we have overlay issues with the input 

![image](https://user-images.githubusercontent.com/4030377/91207310-6b340480-e6bd-11ea-8f8d-0905209a6a8f.png)

Reply:
Got it. In that case, wouldn't `StyledTextInput` start at `z-index: -1` and then apply `z-index: 1` on hover/active?

Reply:
`z-index: -1` begins to place elements behind the body at that point

![image](https://user-images.githubusercontent.com/4030377/91234972-52d9df00-e6e9-11ea-912c-cc87349b61bf.png)

Reply:
This is why resetting the stacking context by applying `z-index: 0` to the `InputGroup` wrapper is so important. Once that is done, the children only need to receive either `z-index: 1` or `z-index: -1`.

Reply:
See https://developer.mozilla.org/en-US/docs/Web/CSS/z-index#Values for details.